### PR TITLE
Refactor ERC4626 tests

### PR DESCRIPTION
**Pull request by joaobrunoah**
# Description

This PR prepares the code to extend E2E ERC4626 Swap tests to other pool types. The current implementation of ERC4626 tests had a lot of redundant code (such as approvals) and didn't use the default "createPool" and "initPool" from BaseVaultTest.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/solidity-utils/test/foundry/utils/BaseTest.sol [hunk 1]
```diff
@@ -124,6 +165,17 @@ abstract contract BaseTest is Test, GasSnapshot {
             deal(address(tokens[i]), user, defaultBalance);
         }
 
+        for (uint256 i = 0; i < erc4626Tokens.length; ++i) {
+            // Give underlying to user, so it can deposit in the wrapped token.
+            ERC20TestToken(erc4626Tokens[i].asset()).mint(user, defaultBalance);
+
+            // Deposit underlying to mint wrapped tokens mto the user.
```
Comment (jubeira) [inline pkg/solidity-utils/test/foundry/utils/BaseTest.sol:172]:
```suggestion
            // Deposit underlying to mint wrapped tokens to the user.
```

File: pkg/vault/test/foundry/BufferDoSProtection.t.sol [hunk 2]
```diff
@@ -172,8 +160,8 @@ contract BufferDoSProtectionTest is BaseVaultTest {
         BaseVaultTest.Balances memory balancesAfter = getBalances(lp, tokens);
 
         // Check wrap results.
-        assertEq(amountIn, _wrapAmount, "AmountIn (underlying deposited) is wrong");
-        assertEq(amountOut, waDAI.previewDeposit(_wrapAmount), "AmountOut (wrapped minted) is wrong");
+        assertGe(amountIn, _wrapAmount, "AmountIn (underlying deposited) is wrong");
+        assertLe(amountOut, waDAI.previewDeposit(_wrapAmount), "AmountOut (wrapped minted) is wrong");
```
Comment (jubeira) [inline pkg/vault/test/foundry/BufferDoSProtection.t.sol:164]: (here and below): can we add a comment explaining this change?

File: pkg/vault/test/foundry/BufferVaultPrimitive.t.sol [hunk 3]
```diff
@@ -788,23 +798,33 @@ contract BufferVaultPrimitiveTest is BaseVaultTest {
...
+            amountIn,
+            _wrapAmount + convertFactorExactOut,
+            convertFactorExactOut,
+            "AmountIn (underlying deposited) is wrong"
+        );
+        assertApproxEqAbs(
             balancesAfter.lpTokens[daiIdx],
-            balancesBefore.lpTokens[daiIdx] - _wrapAmount - convertFactorIn,
+            balancesBefore.lpTokens[daiIdx] - _wrapAmount - convertFactorExactOut,
+            convertFactorExactOut,
```
Comment (jubeira) [inline pkg/vault/test/foundry/BufferVaultPrimitive.t.sol:815]: This one will probably change in upcoming PRs right?

**Review by jubeira — APPROVED** [review]
LGTM, makes sense to move these things to the base layer now that we have tests for this everywhere.

**Comment by joaobrunoah** [comment]
> Mostly comments on docs. Some still have `vaultConvertFactor` - is this complete? I guess they'll be changed in the PRs on top of this one?

Exactly, vaultConvertFactor is fully removed in #967 . This one is just the base for that PR.

File: pkg/vault/test/foundry/BufferVaultPrimitive.t.sol (hunk 3 above)
Comment (joaobrunoah) [reply pkg/vault/test/foundry/BufferVaultPrimitive.t.sol:815]: Yes, exactly! #967 changes it.

**Pull request merged**